### PR TITLE
Registers package with sunpy template

### PR DESCRIPTION
**Pull request by CyclingNinja**

### CODE REVIEW

File: .ruff.toml
Comment (samaloney) [inline .ruff.toml:1]: again our current file has some changes

File: .flake8
[the remark this answers is not in the document]
Comment (Cadair) [reply .flake8:1]: fwiw, we kept this file around in the template just for people's editors to use. We might want to ditch it upstream tbh.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,19 +1,17 @@
 repos:
     # This should be before any formatting hooks like isort
   - repo: https://github.com/astral-sh/ruff-pre-commit
-    rev: "v0.3.5"
+    rev: "v0.5.2"
     hooks:
       - id: ruff
         args: ["--fix"]
-      # Run the formatter.
-      - id: ruff-format
```
[the remark this answers is not in the document]
Comment (CyclingNinja) [reply .pre-commit-config.yaml:9]: Yep, what is it specifically that you're looking at?

**Comment by nabobalis** [comment]
Can this PR be updated with the most recent upstream changes from the template?

File: .pre-commit-config.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Cadair) [reply .pre-commit-config.yaml]: I have re-added ruff-format and configed it to not stand on the template files.

File: pyproject.toml [hunk 2]
```diff
@@ -1,10 +1,64 @@
...
+[project]
+name = "sunkit_spex"
+description = "A package for solar X-ray spectroscopy."
+requires-python = ">=3.9"
+readme = { file = "README.rst", content-type = "text/x-rst" }
+license = { file = "licenses/LICENSE.rst" }
+authors = [
+  { name = "The SunPy Community", email = "sunpy@googlegroups.com" },
+]
+dependencies = [
```
Comment (Cadair) [inline pyproject.toml:18]: Are all these dependences needed in the required list?

**Comment by Cadair** [comment]
I think this should be good for another look from everyone.

File: pyproject.toml (hunk 2 above)
Comment (samaloney) [reply pyproject.toml:18]: See #164

File: sunkit_spex/_dev/__init__.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+"""
+This package contains utilities that are only used when developing drms in a
+copy of the source repository.
```
Comment (samaloney) [inline sunkit_spex/_dev/__init__.py:3]:
```suggestion
This package contains utilities that are only used when developing 
sunkit-spex in a copy of the source repository.
```

**Review by samaloney — APPROVED** [review]
Looks good to me!

**Review by DanRyanIrish — APPROVED** [review]

**Pull request merged**